agent/agentscripts/agentscripts_test.go

```diff
@@ -103,15 +106,44 @@ func TestTimeout(t *testing.T) {
 	t.Parallel()
 	runner := setup(t, nil)
 	defer runner.Close()
+	aAPI := agenttest.NewFakeAgentAPI(t, slogtest.Make(t, nil), nil, nil)
 	err := runner.Init([]codersdk.WorkspaceAgentScript{{
 		LogSourceID: uuid.New(),
 		Script:      "sleep infinity",
 		Timeout:     time.Millisecond,
-	}})
+	}}, aAPI.ScriptCompleted)
 	require.NoError(t, err)
 	require.ErrorIs(t, runner.Execute(context.Background(), nil), agentscripts.ErrTimeout)
 }
 
+func TestScriptReportsTiming(t *testing.T) {
```

Comment:
Nice test 👍 

Related to my comment above, we should be capturing some additional metadata about whether the script blocks startup and therefore we'll need some more nuance in this test.

Reply:
Also: let your code breathe...

Try grouping logically-related lines together to enhance readability and show intent.
Terseness is not a virtue in Go, readability is very much one though.